src/boot/efi/efi-string.c

```diff
@@ -272,6 +274,419 @@ bool efi_fnmatch(const char16_t *pattern, const char16_t *haystack) {
 DEFINE_PARSE_NUMBER(char, parse_number8);
 DEFINE_PARSE_NUMBER(char16_t, parse_number16);
 
+static void format_number(
```

Comment:
Maybe, it is better to introduce tests for this function.

Reply:
Right. But, even so, still I'd like to have tests for the function, though that's not strong requirement.